orderer/consensus/etcdraft/chain.go

```diff
@@ -921,6 +921,40 @@ func (c *Chain) catchUp(snap *raftpb.Snapshot) error {
 		next++
 	}
 
+	if memberChanged { // There is a membership change, need to re-calculate local raft id and refresh communications
+		id, err := detectSelfID(c.opts.Consenters, c.opts.Cert, c.logger)
+		if err != nil { // means i'm not in the channel, halt it
```

Comment:
I still don't understand what we're trying to accomplish here.

1. If all you want is to rotate a certificate of a consenter, you can do it without changing the ID via removing the certificate and adding a new one like demonstrated [here](https://github.com/hyperledger/fabric/blob/master/integration/raft/config_test.go#L441-L578). In such a case, Raft will assign [the old ID](https://github.com/hyperledger/fabric/blob/e4c1fccb442a42ad1d3c9a603d1c9b4fec8c7984/orderer/consensus/etcdraft/membership.go#L81-L86) to the consenter with the rotated certificate.
2. If you want to remove a node then the chain will halt anyway, and you will never reach this code path unless you remove the node, it will not detect its removal and then you change your mind fast enough and re-add it again somehow. 

Since config changes require out of band signature assembly, it's unlikely that this scenario I think you are trying to make available (removal and instant re-addition) will be required, so I don't think the benefits of this change are more than the risk involved in having it.